Sources/Promises/Promise.swift

```diff
@@ -40,14 +40,14 @@ public final class Promise<Value> {
   public convenience init(_ work: @autoclosure () throws -> Value) {
     do {
       let resolution = try work()
-      switch resolution {
-      case let error as NSError:
-        self.init(error)
-      case let objCPromise as ObjCPromise<AnyObject>:
-        self.init(objCPromise)
-      default:
-        self.init(ObjCPromise<AnyObject>.__resolved(with: Promise<Value>.asAnyObject(resolution)))
-      }
+      if type(of: resolution) is NSError.Type {
```

Comment:
Is it possible to go back to a switch here to minimize the change?

Reply:
This is not possible to go back to a switch statement here due to the new type changes